src/main/java/com/incentives/piggyback/offers/serviceImpl/OfferServiceImpl.java

```diff
@@ -109,4 +103,34 @@ public String sendNotification(BroadcastRequest broadcastRequest) {
 		return response.getBody().getData();
 	}
 
+	@Override
+	public List<UserData> getUsersWithInterest(List<Long> users, List<String> interests) {
+		String url = env.getProperty("users.api.usersWithInterest");
+		HttpHeaders headers = new HttpHeaders();
+		headers.set("Accept", MediaType.APPLICATION_JSON_VALUE);
+		UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(url)
+				.queryParam("users", users)
+				.queryParam("interest", interests);
+		HttpEntity<?> entity = new HttpEntity<>(headers);
+		ResponseEntity<List<UserData>> response = 
+				restTemplate.exchange(builder.toUriString(), HttpMethod.GET, 
+						entity, new ParameterizedTypeReference<List<UserData>>(){});
+		if (CommonUtility.isNullObject(response.getBody()) ||
+				CommonUtility.isValidList(response.getBody()))
+			throw new InvalidRequestException("No users with desired interest found!");
+		return response.getBody();
+	}
+	
+
+	private void publishOffer(OfferEntity offer, String status) {
+		messagingGateway.sendToPubsub(
+				CommonUtility.stringifyEventForPublish(
+						gson.toJson(offer),
+						status,
+						Calendar.getInstance().getTime().toString(),
+						"",
```

Comment:
The partnerId must be passed by Offers Service in the event

Reply:
it's already a part of offer entity